collect_app/src/main/java/org/odk/collect/android/formentry/FormEntryMenuDelegate.java

```diff
@@ -111,6 +111,7 @@ public boolean onOptionsItemSelected(MenuItem item) {
             if (audioRecorder.isRecording()) {
                 DialogFragmentUtils.showIfNotShowing(RecordingWarningDialogFragment.class, activity.getSupportFragmentManager());
             } else {
+                formEntryViewModel.updateAnswersForScreen(answersProvider.getAnswers(), false);
```

Comment:
I was thinking about this and I think it was there in the past but you got rid of it in one of your prs. I'm pretty sure that while solving this issue I even found that pull request but now I can't recall what pr it was... or maybe I just had a dream :smile:

Reply:
> I was thinking about this and I think it was there in the past but you got rid of it in one of your prs.

That's probably right! I think it would be good to investigate using `onPause` instead `onSaveInstanceState`. The difference now is that we have more ways to test Activity lifecycle, save points and audit logs so I think we could write a test that checks the log at `onPause` if that is a better way to do it.

Reply:
I guess it must have been contributing to https://github.com/getodk/collect/issues/5291 or https://github.com/getodk/collect/issues/5301. I'd be interested to see if adding it back in again causes either issue to reappear. And again, our ability to test this kind of stuff is in a way better place so it should be more viable to deal with any edge cases.

Reply:
And just for additional context on my thinking here: I do think it's worth doing bigger reworks here now rather than just do the minimal changes to fix the bugs now we have the ability to write better tests.

Reply:
Ok I've investigated using `onPause` or any other method called when an activity is being closed/recreated like `onStop`/`onDestroy` and I'm rather sure we can't use any of those to solve this issue and make the code clearer.
To solve this issue we want to update answers and flush the audit logger to make sure the answer is saved when we navigate to settings. However, we don't want to flush the logger in `onPause`/`onStop`/`onDestroy` because this would save logs for every screen rotation for example. Basically, this would do more harm than good.
If only we had a way to distinguish between activity recreation (when a device is rotated) and real form closing when the activity is killed then we could implement some improvements but in Android it's not easy.

Reply:
> However, we don't want to flush the logger in onPause/onStop/onDestroy because this would save logs for every screen rotation for example.


Ah of course!